coordinator/coordinator.go

```diff
@@ -648,3 +885,103 @@ func (c *Coordinator) executeJob(jobKey string) error {
 	}
 	return nil
 }
+
+type ResourceUpdatedEvent struct {
+	ResourceID metadata.ResourceID
+	Completed  time.Time
+}
+
+func (c *ResourceUpdatedEvent) Serialize() (Config, error) {
+	config, err := json.Marshal(c)
+	if err != nil {
+		panic(err)
+	}
+	return config, nil
+}
+
+func (c *ResourceUpdatedEvent) Deserialize(config Config) error {
+	err := json.Unmarshal(config, c)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+//here we get a notificaiton from a worker that there was a succesful update to a resource
+func (c *Coordinator) signalResourceUpdate(key string, value string) error {
+	c.Logger.Info("Updating metdata with latest resource update status and time", key)
+	s, err := concurrency.NewSession(c.EtcdClient, concurrency.WithTTL(1))
+	if err != nil {
+
+		return err
+	}
+	defer s.Close()
+	mtx, err := c.createJobLock(key, s)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := mtx.Unlock(context.Background()); err != nil {
+			c.Logger.Debugw("Error unlocking mutex:", "error", err)
+		}
+	}()
+	resUpdatedEvent := &ResourceUpdatedEvent{}
+	if err := resUpdatedEvent.Deserialize(Config(value)); err != nil {
+		return err
+	}
+	//TODO set it so an empty string in the new schedule does nothing
```

Comment:
Is this still todo?

Reply:
it's Todone? 0.5 (halfway). Will need to fully fix it in the deployment.